porcupine-core/src/Control/Arrow/FoldA.hs

```diff
@@ -65,7 +65,14 @@ secondP :: (Arrow a) => a c c' -> a (Pair b c) (Pair b c')
 secondP ar =
   arr toTup >>> second ar >>> arr fromTup
 
--- | If @i@ is () and @arr@ is (->), then 'FoldA' is just 'Fold'
+-- | This is a generalization of 'Control.Foldl.Fold' that
+-- allows computing on arrows.
+--
```

Comment:
TODO: motivate why the type of the initial value of the accumulator has been kept separate from the result type.